logic/auth.js

```diff
@@ -40,6 +46,47 @@ function getCachedPassword() {
     return devicePassword;
 }
 
+// Change the device name
+async function changeName(name) {
+    resetChangeNameStatus();
+
+    changeNameStatus.percent = 1; // eslint-disable-line no-magic-numbers
+
+    changeNameStatus.percent = 40; // eslint-disable-line no-magic-numbers
+
+    let complete = false;
+    try {
+        // get user data
+        const user = await diskLogic.readUserFile();
+
+        // update user name
+        user.name = name;
+
+        // update user file
+        await diskLogic.writeUserFile({ ...user });
+
+        complete = true;
+
+        changeNameStatus.percent = 100;
+    } catch (error) {
+        if (error.response.status === constants.STATUS_CODES.UNAUTHORIZED) {
```

Comment:
```suggestion
        if (error.response.status === constants.STATUS_CODES.FORBIDDEN) {
```

Reply:
I've seen the same code as you on the previous function and the thing I don't understand is why use the flag forbidden and respond unauthorised = true?
I've done this by "logic" on my reflexion but I can put it back as you mentioned if it's normal

Reply:
Yeah you're right. 

https://stackoverflow.com/a/6937030

I think we don't need an unauthorised here, because the only error we can get is an fs error (read or write), contrary to the other function which need auth. 

Maybe just return Internal server error if something goes wrong?